index.html

```diff
@@ -4432,11 +4432,16 @@ <h2 id="AudioBufferSourceNode">
                 <code>offset</code> is negative.</span> If <code>offset</code>
                 is greater than <code>loopEnd</code>, playback will begin at
                 <code>loopEnd</code> (and immediately loop to
-                <code>loopStart</code>). This parameter is converted to an
-                exact sample frame offset within the buffer by multiplying by
-                the buffer's sample rate and rounding to the nearest integer
-                value. Thus its behavior is independent of the value of the
-                <a href=
+                <code>loopStart</code>). <code>offset</code> is clamped to [0,
+                <code>duration</code>], where <code>duration</code> is the
+                value of the <code>duration</code> attribute of the
+                <code>AudioBuffer</code> set to the <code>buffer</code>
+                attribute of this <code>AudioBufferSourceNode</code>, 0 if the
+                <code>buffer</code> attribute has not been set. This parameter
```

Comment:
Kind of. I could not find something obvious to do. Maybe it would be better to do the clamping at the time when the audio is actually starting (currentTime + 1000 in your example), and hope that a buffer has been set ? Maybe it would be better ?